server/api.js

```diff
@@ -60,6 +60,71 @@ const cantPlay = (user, tourney) =>
     "Mapper",
   ]);
 
+const canEditWarmup = async (user, playerNo, match) => {
+  async function isCaptainOf(playerName, teamName, tourney) {
+    const team = await Team.findOne({ name: teamName, tourney: tourney }).populate('players');
+    if (!team || !team.players || !team.players[0]) return false;
+    return team.players[0].username === playerName;
+  }
+
+  const tourney = await Tournament.findOne({ code: match.tourney });
+
+  // Admin can always edit warmup
+  if (isAdmin(user, tourney)) return true;
```

Comment:
`isAdmin` expects just the tourney code, not the tourney object from the db
so you can just do `isAdmin(user, match.tourney)` and remove the db fetch

Reply:
Ahh, fixed in the latest commit, db fetch is still required to check if the tourney is a team tourney, so I've moved it back down.